website/docs/r/kinesis_log_source.html.markdown

```diff
@@ -0,0 +1,103 @@
+---
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_log_metrics_source"
+description: |-
+  Provides a Sumologic Kinesis Log source. This source is used to integrate with Metrics Stream via Kinesis Firehose from AWS.
+---
+
+# sumologic_kinesis_log_source
+
+Provides a Sumologic Kinesis Log source. This source is used to ingest log via Kinesis Firehose from AWS.
+
+__IMPORTANT:__ The AWS credentials are stored in plain-text in the state. This is a potential security issue.
+
+## Example Usage
+```hcl
+locals {
+
+  resource "sumologic_kinesis_log_source" "kinesis_log_access_key" {
+    name         = "Kinesis Log"
+    description  = "Description for Kinesis Log Source"
+    category     = "prod/kinesis/log"
+    content_type = "KinesisLog"
+    collector_id = "${sumologic_collector.collector.id}"
+    authentication {
+      type       = "S3BucketAuthentication"
+      access_key = "someKey"
+      secret_key = "******"
+    }
+
+    path {
+      type            = "KinesisLogPath"
+      bucket_name     = "testBucket"
+      path_expression = "http-endpoint-failed/*"
+      scan_interval   = 30000
+    }
+  }
+
+  resource "sumologic_kinesis_log_source" "kinesis_log_role_arn" {
+    name         = "Kinesis Log"
+    description  = "Description for Kinesis Log Source"
+    category     = "prod/kinesis/log"
+    content_type = "KinesisLog"
+    collector_id = "${sumologic_collector.collector.id}"
+
+    authentication {
+      type     = "AWSRoleBasedAuthentication"
+      role_arn = "arn:aws:iam::604066827510:role/cw-role-SumoRole-4AOLS73TGKYI"
+    }
+
+    path {
+      type            = "KinesisLogPath"
+      bucket_name     = "testBucket"
+      path_expression = "http-endpoint-failed/*"
+      scan_interval   = 30000
+    }
+  }
+
+  resource "sumologic_collector" "collector" {
+    name        = "my-collector"
+    description = "Just testing this"
+  }
+}
+```
+
+## Argument reference
+
+In addition to the common properties, the following arguments are supported:
+
+ - `content_type` - (Required) The content-type of the collected data. Details can be found in the [Sumologic documentation for hosted sources][1].
```

Comment:
Why is this a required field if the only accepted type is Kinesis? It probably shouldn't even be exposed for the kinesis resource type

Reply:
In the kinesis metric source file, we also have this field. And here I just use the same list for properties.

Reply:
I don't know if there is any point in exposing this field when there is only one valid value. Not sure who added this for kinesis metric source, but we shouldn't blindly follow the same mistake

Reply:
is there multiple values possible for logs source?

Reply:
For polling source https://github.com/SumoLogic/terraform-provider-sumologic/blob/003ec94c82acce12d20e530b00d073e2c6079dd1/website/docs/r/polling_source.html.markdown, there will be multiple values

It seems to me that `content_type` is a field that will be use in different source types of log source resources. It doesn't seem to me as an issue considering it's a generally used property across all types of sources in sumo world.

Reply:
But here we are dealing with kinesis log source: for this source type, are there multiple values for content type?

We need to have content type for generic source types such as polling source (which is already deprecated as indicated by the link you shared) and HTTP source, because these are both generic source types that accept different content types. However when we make a resource that is of a specific source type (such as kinesis log), if there is only one possible value for a field, then it's redundant to ask user to specify that field

Reply:
It's unfortunate that most of the existing source types in our provider requires this content type to be specified by user even though it's not needed. But I think it's important to recognize the redundancy instead of defending it just because that's how other source types were set up. If we want to remove content type, probably we should remove it everywhere that doesn't require content type, which is a bigger effort

Reply:
@maimaisie I'm reading your comments. Are you fine to leave it as this in this PR with the field `contentType` to keep consistency in format for other sources as the cleanup would also require changes in other source types?

Reply:
Yes, let's go ahead with this for now